src/include/codegen/query_cache.h

```diff
@@ -1,16 +1,46 @@
+//===----------------------------------------------------------------------===//
 //
-// Created by bechob on 4/11/17.
+//                         Peloton
 //
-
-#ifndef PELOTON_CACHE_H
-#define PELOTON_CACHE_H
-
-
-
-class cache {
-
+// value_peeker_proxy.h
+//
+// Identification: src/include/codegen/cache.h
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+namespace peloton {
+namespace codegen {
+class QueryCache {
+public:
+  static QueryCache& Instance() {
+    static  QueryCache instance;
+    return instance;
+  }
+  bool FindPlan(const planner::AbstractPlan & key) {
```

Comment:
Can we combine FindPlan() and GetQuery() instead of having them separately, unless there is any special reason?